### DISCUSSION TITLE
Upgrading operator versions

### DISCUSSION THREAD

**Discussion by ananthk01** [strimzi/strimzi-kafka-operator#4663] [Q&A] [answered]
I am looking to upgrade the strimzi operator version from 0.20 to 0.21 and then once 0.23 is available then to 0.22. What's the advice here? In general should every version (in this case 0.20.1) be applied before moving to the target version or can 0.21 be applied directly?

**Comment by scholzj** [comment, marked as answer]
> I am looking to upgrade the strimzi operator version from 0.20 to 0.21 and then once 0.23 is available then to 0.22.

I'm not sure I follow this logic: `from 0.20 to 0.21 and then once 0.23 is available then to 0.22`. 

> What's the advice here? In general should every version (in this case 0.20.1) be applied before moving to the target version or can 0.21 be applied directly?

The 0.22 version is something what you would need to install because of the CRD migration to `apiextensions/v1`. Apart from that you do not need to install every single version. You should for example be able to go from 0.20 directly to 0.22.

**Reply by ananthk01** [reply]
The upgrades are typically planned one version behind at this time. That's the logic there. 

Thanks for the explanation, Jakub. Because of what's happening with 0.22, is the advice here to move directly to 0.22 instead of 0.21?

**Reply by scholzj** [reply]
> is the advice here to move directly to 0.22 instead of 0.21?

No, not really. I think most other users would normally jump directly to 0.22 just to save the effort of doing two upgrades. But if you prefer to be always one version behind it is fine ... you can go now to 0.21 and later to 0.22 ... there is no known problem with it or anything like that.

**Reply by ananthk01** [reply]
Great. Thanks!
